src/ProjectTemplates/Microsoft.Extensions.AI.Templates/src/ChatWithCustomData/ChatWithCustomData.Web/Components/Pages/Chat/Chat.razor

```diff
@@ -60,9 +60,16 @@
         // Display a new response from the IChatClient, streaming responses
         // aren't supported because Ollama will not support both streaming and using Tools
         currentResponseCancellation = new();
-        var response = await ChatClient.GetResponseAsync(messages, chatOptions, currentResponseCancellation.Token);
-        currentResponseMessage = response.Message;
-        ChatMessageItem.NotifyChanged(currentResponseMessage);
+        try
+        {
+            var response = await ChatClient.GetResponseAsync(messages, chatOptions, currentResponseCancellation.Token);
+            currentResponseMessage = response.Message;
+            ChatMessageItem.NotifyChanged(currentResponseMessage);
+        }
+        catch (HttpRequestException ex) when (ex.StatusCode == System.Net.HttpStatusCode.NotFound)
+        {
+            throw new Exception("Failed to generate embeddings. Your workstation may need additional setup, see the README for details.", ex);
+        }
```

Comment:
I think it would be better if we don't have this try/catch. If it fails, there will be an exception anyway. The underlying client implementation is better placed than we are to interpret what the error was and supply the most helpful explanation to guide developers to a solution. If they are doing a bad job with the underlying exception, we should help them out with improving that.

From experience, any time we wrap an error case like this, users interpret the error as a problem in our code rather than the underlying cause.

Reply:
Makes sense to me, I was really wanting to give an extra nudge to look at the README, but I can see your point (below) about references to the README won't make sense once deployed/over the life of the project. I'll drop these, thanks!